src/System.Web.OData/OData/Query/ODataQueryOptions.cs

```diff
@@ -87,6 +87,11 @@ public ODataQueryOptions(ODataQueryContext context, HttpRequestMessage request)
         }
 
         /// <summary>
+        /// Gets Column order.
+        /// </summary>
+        public static Dictionary<string, int> ColumnOrder { get; set; }
```

Comment:
Change to private set and initialize to new Dictionary<string, int>() in the constructor. The code analysis error says to make this read-only. You don't need to let another class set if equally to a new dictionary, only to populate it which can be achieved with a read-only property.